src/Crawler/RepositoryFilesDeserializer.cs

```diff
@@ -0,0 +1,70 @@
+﻿using ModelRepoBrowser.Crawler.XmlModels;
+using System.Xml;
+using System.Xml.Serialization;
+
+namespace ModelRepoBrowser.Crawler
+{
+    public static class RepositoryFilesDeserializer
+    {
+        private static Dictionary<Type, XmlSerializer> serializers = new();
+
+        private static T? DeserializeDatasection<T>(Stream stream)
+        {
+            var serializer = serializers.GetValueOrDefault(typeof(T), new XmlSerializer(typeof(T)));
+            using (var xmlReader = new XmlTextReader(stream))
+            {
+                xmlReader.Namespaces = false;
+                xmlReader.ReadToDescendant("DATASECTION");
+                var subtree = xmlReader.ReadSubtree();
+
+                return (T?)serializer.Deserialize(subtree);
+            }
+        }
+
+        /// <summary>
+        /// Parse an ilisite.xml from an <paramref name="xmlStream"/>.
+        /// </summary>
+        /// <param name="xmlStream">The XML-<see cref="Stream"/> that contains an ilisite.xml.</param>
+        /// <returns>The parsed <see cref="Site"/> object of the ilisite.xml.</returns>
+        /// <exception cref="InvalidOperationException">Could not deserialize the provided XML.</exception>
+        internal static Site? ParseIliSite(Stream xmlStream)
+        {
+            var dataSection = DeserializeDatasection<IliSitesDataSection>(xmlStream);
+            return dataSection?.SiteMetadata?.Site;
+        }
+
+        /// <summary>
+        /// Parse an ilimodels.xml from an <paramref name="xmlStream"/>.
+        /// </summary>
+        /// <param name="xmlStream">The XML-<see cref="Stream"/> that contains an ilimodels.xml.</param>
+        /// <returns>Flat list of all parsed ModelsMetadata contained in ilimodels.xml.</returns>
+        /// <exception cref="InvalidOperationException">Could not deserialize the provided XML.</exception>
+        internal static IEnumerable<ModelMetadata> ParseIliModels(Stream xmlStream)
+        {
+            var dataSection = DeserializeDatasection<IliModelsDatasection>(xmlStream);
+            var result = dataSection?.Items
+                                .Where(x => x is not null && x.ModelMetadata is not null)
```

Comment:
PP: Kann vereinfacht werden zu 
```suggestion
                                .Where(x => x?.ModelMetadata is not null)
```